src/python/WMCore/Services/pycurl_manager.py

```diff
@@ -255,19 +243,21 @@ def set_opts(self, curl, url, params, headers,
         if self.tmgr:
             token = self.tmgr.getToken()
             if token:
-                headers['Authorization'] = 'Bearer {}'.format(token)
+                thisHeaders['Authorization'] = 'Bearer {}'.format(token)
 
         if verb in ('POST', 'PUT'):
             # only these methods (and PATCH) require this header
             thisHeaders["Content-Length"] = str(len(encoded_data))
 
-        # we must pass url as a string data-type, otherwise pycurl will fail with error
+        # we must pass url as a bytes data-type, otherwise pycurl will fail with error
         # TypeError: invalid arguments to setopt
         # see https://curl.haxx.se/mail/curlpython-2007-07/0001.html
         curl.setopt(pycurl.URL, encodeUnicodeToBytes(url))
-        if thisHeaders:
-            curl.setopt(pycurl.HTTPHEADER, \
-                [encodeUnicodeToBytes("%s: %s" % (k, v)) for k, v in viewitems(thisHeaders)])
+        # In order to enable service intercommunication with compressed HTTP body,
+        # we need to enable this header here, in case it has not been provided by upstream.
+        thisHeaders.setdefault("Accept-Encoding", "gzip")
+        curl.setopt(pycurl.HTTPHEADER, [encodeUnicodeToBytes("%s: %s" % (k, v)) for k, v in viewitems(thisHeaders)])
```

Comment:
I doubt it is correct syntax to set the header, as far as I recall it should be either explicit like this:
```
curl.setopt(pycurl.ACCEPT_ENCODING, 'gzip')
```
or doing this way:
```
curl.setopt(pycurl.HTTPHEADER, ['Transfer-Encoding: chunked'])
```
There is no need to convert string to bytes the header. I think the final answer depends on how pycurl implements this internally, therefore I'm not sure that what you do is correct syntax (it can be though).

Reply:
Please read the documentation here:
https://curl.se/libcurl/c/CURLOPT_ACCEPT_ENCODING.html

but in short, there are two ways to set "Accept-Encoding" http header with pycurl:
* in the HTTPHEADER option, as any other header.
* in the ACCEPT_ENCODING option, but in this case it also enables the automatic data decompression in libcurl, whenever a `Content-Enconding` response header is returned and the compression algorithm is supported by the libcurl build.